src/azure-cli/azure/cli/command_modules/util/tests/latest/test_rest.py

```diff
@@ -14,8 +14,50 @@
 
 class ResourceGroupScenarioTest(ScenarioTest):
 
+    def test_rest_arm(self):
+        from knack.util import CLIError
+
+        self.kwargs.update({
+            'rg': self.create_random_name("test-rest-rg", length=20)
+        })
+
+        # Test ARM Subscriptions - List
+        # https://docs.microsoft.com/en-us/rest/api/resources/subscriptions/list
+        self.cmd('az rest -u /subscriptions?api-version=2020-01-01',
+                 checks=[self.exists("value")])
+
+        # Test ARM Tenants - List
+        # https://docs.microsoft.com/en-us/rest/api/resources/tenants/list
+        self.cmd('az rest -u /tenants?api-version=2020-01-01',
+                 checks=[self.exists("value")])
+
+        # Resource Groups - Create Or Update
+        # https://docs.microsoft.com/en-us/rest/api/resources/resourcegroups/createorupdate
+        self.cmd('az rest -m PUT -u https://management.azure.com/subscriptions/{{subscriptionId}}/resourcegroups/{rg}?api-version=2019-10-01 '
+                 '--body \'{{"location": "eastus"}}\'',
+                 checks=[self.check("name", '{rg}')])
+
+        # Resource Groups - Get
+        # https://docs.microsoft.com/en-us/rest/api/resources/resourcegroups/get
+        self.cmd('az rest -u https://management.azure.com/subscriptions/{{subscriptionId}}/resourcegroups/{rg}?api-version=2019-10-01',
+                 checks=[self.check("name", '{rg}')])
+
+        # Resource Groups - Delete
+        # https://docs.microsoft.com/en-us/rest/api/resources/resourcegroups/delete
+        self.cmd('az rest -m DELETE -u https://management.azure.com/subscriptions/{{subscriptionId}}/resourcegroups/{rg}?api-version=2019-10-01',
+                 checks=[])
+
+        # Resource Groups - Get
+        # Polling for 404
+        # TODO: return 3 for 404
+        with self.assertRaises(CLIError):
+            while True:
+                time.sleep(5)
+                result = self.cmd('az rest -u https://management.azure.com/subscriptions/{{subscriptionId}}/resourcegroups/{rg}?api-version=2019-10-01',
+                                  checks=[])
```

Comment:
Could we use the command `az group wait -n {group-name} --deleted --timeout {timeout} ` to monitor the deletion status of resource group instead of manually polling?

Reply:
This test is simply for `az rest`. It's better to keep it limited to `az rest`.

Reply:
OK~